source/include/shadow.h

```diff
@@ -115,11 +117,37 @@ typedef enum ShadowStatus
  */
 #define SHADOW_PREFIX_LENGTH              ( ( uint16_t ) ( sizeof( SHADOW_PREFIX ) - 1U ) )
 
+/**
+ * @ingroup shadow_constants
+ * @brief The root of all un-named "Classic" Shadow MQTT topics
+ *        from here https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-mqtt.html.
+ */
+#define SHADOW_CLASSIC_ROOT               "/shadow"
+
+/**
+ * @ingroup shadow_constants
+ * @brief The length of #SHADOW_CLASSIC_ROOT.
+ */
+#define SHADOW_CLASSIC_ROOT_LENGTH        ( ( uint16_t ) ( sizeof( SHADOW_CLASSIC_ROOT ) - 1U ) )
+
+/**
+ * @ingroup shadow_constants
+ * @brief The common root of all named Shadow MQTT topics
+ *        from here https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-mqtt.html.
+ */
+#define SHADOW_NAMED_ROOT                 "/shadow/name/"
```

Comment:
Does this need a slash infront?

Reply:
With the way I've constructed the macros, yes. Obviously I could handle the slash (and extra +1 in the length) if you prefer it that way? Current macros:

```
#define SHADOW_NAMED_ROOT                 "/shadow/name/"

#define SHADOW_NAMED_ROOT_LENGTH          ( ( uint16_t ) ( sizeof( SHADOW_NAMED_ROOT ) - 1U ) )

#define SHADOW_TOPIC_LENGTH( operationLength, suffixLength, thingNameLength, shadowNameLength ) \
    ( operationLength + suffixLength + thingNameLength + shadowNameLength +                     \
      SHADOW_PREFIX_LENGTH +                                                                    \
      ( ( shadowNameLength > 0 ) ? SHADOW_NAMED_ROOT_LENGTH : SHADOW_CLASSIC_ROOT_LENGTH ) )

#define SHADOW_TOPIC_STRING( thingName, shadowName, operation, suffix )           \
    ( ( sizeof( shadowName ) > 1 ) ?                                              \
      ( SHADOW_PREFIX thingName SHADOW_NAMED_ROOT shadowName operation suffix ) : \
      ( SHADOW_PREFIX thingName SHADOW_CLASSIC_ROOT operation suffix ) )
```

Reply:
No further comment, so counting as resolved. For now at least.